elyra/airflow/bootstrapper.py

```diff
@@ -376,8 +382,18 @@ def package_list_to_dict(cls, filename: str) -> dict:
                         package_name, package_version = line.strip("\n").split(sep=" @ ")
                     elif "===" in line:
                         package_name, package_version = line.strip("\n").split(sep="===")
-                    else:
+                    elif "==" in line:
                         package_name, package_version = line.strip("\n").split(sep="==")
+                    elif line.startswith("-e ") or line.startswith("--editable "):
+                        package_name = line.strip("\n").replace("-e ", "").replace("--editable ", "")
+                        if "#egg=" in package_name:  # editable package from version control system
+                            package_name = package_name.split("=")[-1]
+                        elif "/" in package_name:  # editable package from local directory
+                            package_name = os.path.basename(package_name)
```

Comment:
Still not sure if we should do this because there is no requirement that the directory name matches the package name.

Reply:
We could remove this `if/elif` block and issue a warning message about editable packages directly in this block, but that might result in unnecessary noise since it will be called for both Elyra and current requirements.

Reply:
Thinking this through again, I now agree that it's probably best to leave it as is because it's the only chance to detect whether there are any potential conflicts for the _required_ packages.